cli/src/config/templates.toml

```diff
@@ -16,10 +16,10 @@ commit_summary = 'format_commit_summary_with_refs(self, bookmarks)'
 
 annotate_commit_summary = '''
 separate(" ",
-  format_short_id(change_id),
-  format_short_id(commit_id),
-  format_short_signature(author),
-  format_timestamp(committer.timestamp()),
+  change_id.shortest(8),
+  commit_id.shortest(8),
+  pad_end(8, truncate_end(8, author.username())),
+  committer.timestamp().local().format("%Y-%m-%d"),
```

Comment:
I would include the hour and minute, in case you are looking at a draft PR stack you recently wrote.

Reply:
I would add `%H:%M:%S` then. Consistency is more important than saving 3 characters.

Reply:
I'm not sure. Would you ever want to see the seconds in the blame view? I'd only want that if I did several commits in less than a minute, which is rare.

But I'll leave it up to you and others.

Reply:
I personally don't want to see `%H:%M:%S` at all. I wouldn't even be interested in `%d` (and author as I said before.) I'm just saying that we should stick to the default formatting than adding a couple of variants that look quite similar.

Reply:
It's an interesting question about showing the author. We decided to not call it `jj [file] blame` because we don't want it to seem like a tool for blaming people (at least I think that's why). Removing the author from the output seems compatible with that line of thinking.

OTOH, it does seem a bit useful to help remind you of the context in which a line change. I personally often remember (think I) remember who added a feature, so filtering by user is sometimes helpful at least for finding a commit I'm looking for. But it probably is much less helpful in `jj file annotate` output. I think I just look for the change id (or commit id in git/hg) and date (in addition to the line contents, of course).

So, I think I would personally be happy to try including only change id and date (not even time). But that's just my feeling.

Reply:
(Non-actionable) The actionable part of your comments is pretty clear, but I am a little confused by some details in your comments. I guess I don't follow how you use the annotations?

Yuya, you just want to see the year and the month? So, you only care whether some lines are much older than others? 

Martin, I'm not sure what you are saying in your second paragraph. Where are you "filtering by user" if not in the `jj file annotate` output? The log? I find showing the author useful for precisely the reason you described, for remembering what a commit was about.

Possibly more actionable, but also fanciful and not for this PR: This discussion makes me wonder whether we should include 20-40 characters of the commit description instead of the author and date. I don't know if it would work in practice; I wish there was a way to shorten commit descriptions to X characters. Perhaps this will be `jj`'s first use of ChatGPT 😛 .

Reply:
> Where are you "filtering by user" if not in the jj file annotate output? The log?

Yes, that's what I meant. Sorry that I wasn't clear.

Reply:
> BTW, author name is more visually distinct than change/commit ids and dates, so it will help separate lines originating from different commits.

OTOH, that may make it harder to notice when some nearby lines are from different commits if they have the same author.

Reply:
If this command has machine output, all the details can be output (JSON ?).

Reply:
> If this command has machine output, all the details can be output (JSON ?).

No, the output is meant for humans. It could also be used by scripts, but they probably only need the commit id and the line number.

> It could be a TUI opened by e.g. jj file annotate --interactive.

Yes, a TUI would be very nice. Among other nice things, it could have a binding to hide/show author or date, which would make the defaults less important.

I generally use `tig blame` for annotations, which is quite a nice UI, though I'd change a bunch of things about it (easier and more discoverable key bindings, add the show/hide author or date binding, ...).

Reply:
Which one do we like as the default?
(I'm not going to discuss about the truncation width and other minor things, which can be changed later as needed.)

```toml
# A
annotate_commit_summary = '''
separate(" ",
  change_id.shortest(8),
  committer.timestamp().local().format('%Y-%m-%d'),
)
'''

# B
annotate_commit_summary = '''
separate(" ",
  change_id.shortest(8),
  pad_end(8, truncate_end(8, author.username())),
  committer.timestamp().local().format('%Y-%m-%d'),
)
'''

# C (this patch)
annotate_commit_summary = '''
separate(" ",
  change_id.shortest(8),
  commit_id.shortest(8),
  pad_end(8, truncate_end(8, author.username())),
  committer.timestamp().local().format('%Y-%m-%d'),
)
'''

# D
annotate_commit_summary = '''
separate(" ",
  change_id.shortest(8),
  commit_id.shortest(8),
  pad_end(8, truncate_end(8, author.email())),
  committer.timestamp().local().format('%Y-%m-%d %H:%M:%S'),
)
'''
```

I personally like A or B.

Reply:
I like B with the time added.

Reply:
Ok, updated to B with '%Y-%m-%d %H:%M:%S'.

If there aren't strong complaints, I'll merge this.